R/guides-axis.r

```diff
@@ -1,4 +1,157 @@
+#'   you probably want.
+#' @param n_dodge The number of rows (for vertical axes) or columns (for
+#'   horizontal axes) that should be used to render the labels. This is
+#'   useful for displaying labels that would otherwise overlap.
+#' @param order Used to determine the order of the guides (left-to-right,
+#'   top-to-bottom), if more than one  guide must be drawn at the same location.
+#' @param position Where this guide should be drawn: one of top, bottom,
+#'   left, or right.
+#'
+#' @export
+#'
+#' @examples
+#' # plot with overlapping text
+#' p <- ggplot(mpg, aes(cty * 100, hwy * 100)) +
+#'   geom_point() +
+#'   facet_wrap(vars(class))
+#'
+#' # axis guides can be customized in the scale_* functions or
+#' # using guides()
+#' p + scale_x_continuous(guide = guide_axis(n_dodge = 2))
+#' p + guides(x = guide_axis(angle = 90))
+#'
+#' # can also be used to add a duplicate guide
+#' p + guides(x = guide_axis(n_dodge = 2), y.sec = guide_axis())
+#'
+#'
+guide_axis <- function(title = waiver(), check.overlap = FALSE, angle = NULL, n_dodge = 1,
+                       order = 0, position = waiver()) {
+  structure(
+    list(
+      title = title,
+
+      # customizations
+      check.overlap = check.overlap,
+      angle = angle,
+      n_dodge = n_dodge,
+
+      # general
+      order = order,
+      position = position,
+
+      # parameter
+      available_aes = c("x", "y"),
+
+      name = "axis"
+    ),
+    class = c("guide", "axis")
+  )
+}
+
+#' @export
+guide_train.axis <- function(guide, scale, aesthetic = NULL) {
+
+  aesthetic <- aesthetic %||% scale$aesthetics[1]
+  breaks <- scale$get_breaks()
+
+  empty_ticks <- new_data_frame(
+    list(aesthetic = numeric(0), .value = numeric(0), .label = character(0))
+  )
+  names(empty_ticks) <- c(aesthetic, ".value", ".label")
+
+  if (length(intersect(scale$aesthetics, guide$available_aes)) == 0) {
+    warning(
+      "axis guide needs appropriate scales: ",
+      paste(guide$available_aes, collapse = ", "),
+      call. = FALSE
+    )
+    guide$key <- empty_ticks
+  } else if (length(breaks) == 0) {
+    guide$key <- empty_ticks
+  } else {
+    ticks <- new_data_frame(setNames(list(scale$map(breaks)), aesthetic))
+    ticks$.value <- breaks
+    ticks$.label <- scale$get_labels()
+
+    if (is.list(ticks$.label)) {
+      if (any(sapply(ticks$.label, is.language))) {
+        ticks$.label <- do.call(expression, ticks$.label)
+      } else {
+        ticks$.label <- unlist(ticks$.label)
+      }
+    }
+
+    guide$key <- ticks
+  }
+
+  guide$name <- paste0(guide$name, "_", aesthetic)
+  guide$hash <- digest::digest(list(guide$title, guide$key$.value, guide$key$.label, guide$name))
+  guide
+}
+
+#' @export
+guide_transform.axis <- function(guide, coord, panel_params) {
+  if (is.null(guide$position) || nrow(guide$key) == 0) {
+    return(guide)
+  }
+
+  aesthetics <- names(guide$key)[!grepl("^\\.", names(guide$key))]
+
+  if (all(c("x", "y") %in% aesthetics)) {
```

Comment:
I think this needs to be switched around, no? you want to check if there is no other aesthetics than `x` and `y`, right?